config/config.go

```diff
@@ -161,6 +166,177 @@ type AuthConfig struct {
 	AppendMetadataHeaders bool          `yaml:"append_metadata_headers,omitempty" json:"append_metadata_headers,omitempty"`
 }
 
+// EnvConfigs contains environment configs or URIs to them.
+type EnvConfigs struct {
+	// A list of strings containing environment config URIs
+	// Only the local file system is supported currently, e.g., file://path/to/config.yaml
+	ConfigURIs []string `yaml:"config_uris,omitempty" json:"config_uris,omitempty"`
+
+	// A list of environment configs
+	Inline []EnvironmentConfig `yaml:"inline,omitempty" json:"inline,omitempty"`
+}
+
+// EnvironmentConfig is an Apigee Environment-level config for
+// Envoy Adapter. It contains a list of operations for the adapter to
+// perform request authentication and authorization.
+type EnvironmentConfig struct {
+	// Unique ID of the environment config
+	ID string `yaml:"id" json:"id"`
+
+	// A list of proxy configs
+	ProxyConfigs []ProxyConfig
+}
+
+// ProxyConfig has the proxy configuration.
+type ProxyConfig struct {
```

Comment:
Are we intentionally omitting the "default" settings (target, authn, authz)?

Reply:
I would like each `APIOperation` to be self-contained here. But it's open for debate.

Reply:
Per offline discussion. I put the global target, authn and authz.